helper/cli.c

```diff
@@ -255,25 +355,224 @@ static int cmd_call_odp_sys_info_print(struct cli_def *cli,
+				    char *argv[], int argc)
+{
+	odp_pktio_t hdl = get_pktio(cli, argv, argc);
+
+	if (hdl == ODP_PKTIO_INVALID)
+		return CLI_ERROR;
+
 	odp_pktio_print(hdl);
 
 	return CLI_OK;
 }
 
+static int cmd_call_odp_pktio_stats_print(struct cli_def *cli,
+					  const char *command ODP_UNUSED,
+					  char *argv[], int argc)
+{
+	odp_pktio_t hdl = get_pktio(cli, argv, argc);
+
+	if (hdl == ODP_PKTIO_INVALID)
+		return CLI_ERROR;
+
+	odp_pktio_stats_t stats;
+
+	if (odp_pktio_stats(hdl, &stats) < 0) {
+		cli_error(cli, "%% Unable to query stats.");
+		return CLI_ERROR;
+	}
+
+	(void)cli_log(ODP_LOG_PRINT, "Pktio statistics\n----------------\n");
+	(void)cli_log(ODP_LOG_PRINT, "  in_octets: %" PRIu64 "\n", stats.in_octets);
+	(void)cli_log(ODP_LOG_PRINT, "  in_packets: %" PRIu64 "\n", stats.in_packets);
+	(void)cli_log(ODP_LOG_PRINT, "  in_ucast_pkts: %" PRIu64 "\n", stats.in_ucast_pkts);
+	(void)cli_log(ODP_LOG_PRINT, "  in_mcast_pkts: %" PRIu64 "\n", stats.in_mcast_pkts);
+	(void)cli_log(ODP_LOG_PRINT, "  in_bcast_pkts: %" PRIu64 "\n", stats.in_bcast_pkts);
+	(void)cli_log(ODP_LOG_PRINT, "  in_discards: %" PRIu64 "\n", stats.in_discards);
+	(void)cli_log(ODP_LOG_PRINT, "  in_errors: %" PRIu64 "\n", stats.in_errors);
+	(void)cli_log(ODP_LOG_PRINT, "  out_octets: %" PRIu64 "\n", stats.out_octets);
+	(void)cli_log(ODP_LOG_PRINT, "  out_packets: %" PRIu64 "\n", stats.out_packets);
+	(void)cli_log(ODP_LOG_PRINT, "  out_ucast_pkts: %" PRIu64 "\n", stats.out_ucast_pkts);
+	(void)cli_log(ODP_LOG_PRINT, "  out_mcast_pkts: %" PRIu64 "\n", stats.out_mcast_pkts);
+	(void)cli_log(ODP_LOG_PRINT, "  out_bcast_pkts: %" PRIu64 "\n", stats.out_bcast_pkts);
+	(void)cli_log(ODP_LOG_PRINT, "  out_discards: %" PRIu64 "\n", stats.out_discards);
+	(void)cli_log(ODP_LOG_PRINT, "  out_errors: %" PRIu64 "\n\n", stats.out_errors);
+
+	return CLI_OK;
+}
+
+static int cmd_call_odp_pktio_queue_stats_print(struct cli_def *cli,
+						const char *command ODP_UNUSED,
+						char *argv[], int argc)
+{
+	odp_pktio_t hdl = get_pktio(cli, argv, argc);
+
+	if (hdl == ODP_PKTIO_INVALID)
+		return CLI_ERROR;
+
+	int in_q_cnt = odp_pktin_queue(hdl, NULL, 0);
+
+	if (in_q_cnt > 0) {
+		odp_pktin_queue_t in_qs[in_q_cnt];
+		odp_pktin_queue_stats_t in_stats;
+
+		in_q_cnt = odp_pktin_queue(hdl, in_qs, in_q_cnt);
+
+		(void)cli_log(ODP_LOG_PRINT, "Pktin queue statistics\n----------------------\n");
+
+		for (int i = 0; i < in_q_cnt; i++) {
+			(void)cli_log(ODP_LOG_PRINT, "Pktin queue: %d:\n", i);
+
+			if (odp_pktin_queue_stats(in_qs[i], &in_stats) < 0) {
+				(void)cli_log(ODP_LOG_PRINT,
+					      "  (Unable to read statistics, skipping)\n");
+				continue;
+			}
+
+			(void)cli_log(ODP_LOG_PRINT, "  octets: %" PRIu64 "\n", in_stats.octets);
+			(void)cli_log(ODP_LOG_PRINT, "  packets: %" PRIu64 "\n", in_stats.packets);
+			(void)cli_log(ODP_LOG_PRINT, "  discards: %" PRIu64 "\n",
+				      in_stats.discards);
+			(void)cli_log(ODP_LOG_PRINT, "  errors: %" PRIu64 "\n", in_stats.errors);
```

Comment:
This could be a function. cli_log_pktin_queue_stats(odp_pktin_queue_stats_t *) or something like that. And the same for odp_pktout_queue_stats_t.

Reply:
v4.